### PR TITLE
changing from metric to tagset

### PR DESCRIPTION
**Pull request by jeeyoungk**
As discussed today, i'm changing the `MetricKey` to `TagSet` and moving the human-readable name to `SeriesList`.

@achow @Nathan-Fenner

### CODE REVIEW

File: api/types.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func (tr Timerange) Slots() int {
 // Timeseries is a single time series, identified with the associated tagset.
 type Timeseries struct {
 	Values []float64
-	Metric TaggedMetric // TODO (jee) - change this to use TagSet instead of TaggedMetric.
+	TagSet TagSet
```
Comment (jeeyoungk) [inline api/types.go]: this is the change i want to have - everything else is there to support this change.

File: api/types.go [hunk 2]
```diff
@@ -174,6 +174,7 @@ const (
 type SeriesList struct {
 	Series    []Timeseries
 	Timerange Timerange
+	Name      string // human-readable description of the given time series.
```
Comment (jeeyoungk) [inline api/types.go]: not being used in many different places - but we should be using it in future.

**Comment by achow** [comment]
A bit surprised that Name's equivalent isn't actually used anywhere. LGTM though.

**Comment by Nathan-Fenner** [comment]
LGTM too

**Pull request merged**